genie-core/src/main/java/com/netflix/genie/core/jpa/repositories/JpaJobRequestRepository.java

```diff
@@ -33,10 +34,17 @@
 public interface JpaJobRequestRepository extends JpaRepository<JobRequestEntity, String>, JpaSpecificationExecutor {
 
     /**
-     * Delete all job requests created before the given date.
+     * Returns all job requests created before the given date.
      *
-     * @param date The date before which all job requests should be deleted.
-     * @return The number of deleted records
+     * @param date The date before which all job requests were created.
+     * @return List of job requests
      */
-    Long deleteByCreatedBefore(@NotNull final Date date);
+    List<JobRequestEntity> findByCreatedBefore(@NotNull final Date date);
```

Comment:
As discussed my only concern here is memory if this list becomes too large. Currently this method is called daily as part of a task so we should be ok but it's something we might want to improve going forward either by returning just the Id's or paginating this list.